pkg/apis/serving/v1alpha2/predictor.go

```diff
@@ -79,6 +89,10 @@ func (p *PredictorSpec) Validate(config *InferenceServicesConfig) error {
 	if err := validateReplicas(p.MinReplicas, p.MaxReplicas); err != nil {
 		return err
 	}
+	if err := validateResourceRequirement(p.GetResourceRequirement()); err != nil {
```

Comment:
in fact, requirements *v1.ResourceRequirements doesn't only include CPU and Memory, you can refer to or use below method directly.
k8s.io/kubernetes/pkg/apis/core/validation/validation.go:ValidateResourceRequirements

Reply:
Thanks Hougang, 
I thought there's only CPU or memory from all the samples but in fact it can be anything.  I agree and will do the change.

Reply:
The package k8s.io/kubernetes/pkg/apis/core/validation in fact is not introduced so I tried to introduce it by adding it to import of predictor.go and then dep ensure add. But it always fail and the message is like:
➜  kfserving git:(add_resource_validation) ✗ dep status
Gopkg.lock is out of sync with imports and/or Gopkg.toml. Run dep check for details.
PROJECT            MISSING PACKAGES
k8s.io/kubernetes  [k8s.io/kubernetes/pkg/apis/core/validation]
input-digest mismatch

Then I delete the Gopkg.lock  and whole vendor directory and try to "dep ensure" again, it failed with message:

Solving failure: No versions of sigs.k8s.io/controller-runtime met constraints:
................
	v0.2.0-alpha.1: Could not introduce sigs.k8s.io/controller-runtime@v0.2.0-alpha.1, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	v0.2.0-alpha.0: Could not introduce sigs.k8s.io/controller-runtime@v0.2.0-alpha.0, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	master: Could not introduce sigs.k8s.io/controller-runtime@master, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	actions-test: Could not introduce sigs.k8s.io/controller-runtime@actions-test, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	admissionwebhook: Could not introduce sigs.k8s.io/controller-runtime@admissionwebhook, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	admissionwebhook-1.11: Could not introduce sigs.k8s.io/controller-runtime@admissionwebhook-1.11, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	bulk_deposit: Could not introduce sigs.k8s.io/controller-runtime@bulk_deposit, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	feature/modules: Could not introduce sigs.k8s.io/controller-runtime@feature/modules, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	gh-cicd-test: Could not introduce sigs.k8s.io/controller-runtime@gh-cicd-test, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	release-0.1: Could not introduce sigs.k8s.io/controller-runtime@release-0.1, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	release-0.2: Could not introduce sigs.k8s.io/controller-runtime@release-0.2, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	release-0.3: Could not introduce sigs.k8s.io/controller-runtime@release-0.3, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.
	review: Could not introduce sigs.k8s.io/controller-runtime@review, as it is not allowed by constraint 0.1.9 from project github.com/kubeflow/kfserving.

I think the Gopkg.toml is outdated, But I'm not sure what version we should use and which kubernetes version we are going to support for now, any ideas about this?

Reply:
@hougangliu

Reply:
> Thanks Hougang,
> I thought there's only CPU or memory from all the samples but in fact it can be anything. I agree and will do the change.

I can be GPU resource as well, see here https://github.com/kubeflow/kfserving/blob/master/docs/samples/accelerators/tensorflow-gpu.yaml